### ISSUE TITLE
Rule LC0054 - Interface should start with captial I - Proposal: Ignore spaces between affix and name

### ISSUE THREAD

**Issue by pri-kise** [StefanMaron/BusinessCentral.LinterCop#664]
In general I like the new idea for the Interfaces.

```al
interface "XXX-PTE ICustomInterface"
```

I have the following problem with this rule:
Currenlty our projects are setup with the following settings:

Simplified: settings.json
```json
{
  "al.codeAnalyzers": [
    "${AppSourceCop}",
    "${CodeCop}",
    "${PerTenantExtensionCop}",
    "${UICop}",
    "${analyzerFolder}BusinessCentral.LinterCop.dll"
  ],
  "al.enableCodeActions": true,
  "al.enableCodeAnalysis": true,
  "alOutline.useBCLinterCopCaseRules": true,
  "editor.codeActionsOnSave": {
    "source.fixAll": "explicit"
  },
  "editor.formatOnSave": true,
  "CRS.FileNamePattern": "<ObjectNameShort>.<ObjectTypeShortPascalCase>.al",
  "CRS.FileNamePatternExtensions": "<ObjectNameShort>.<ObjectTypeShortPascalCase>.al",
  "CRS.FileNamePatternPageCustomizations": "<ObjectNameShort>.<ObjectTypeShortPascalCase>.al",
  "CRS.ObjectNamePrefix": "XXX-PTE ",
  "CRS.OnSaveAlFileAction": "Rename",
  "CRS.RemovePrefixFromFilename": true,
}

```

AppSourceCop.json:
```json
{
  "supportedCountries": [
    "DE"
  ],
  "mandatoryAffixes": ["XXX-PTE","XXX_PTE", "PTE_XXX"]
}
```

The problem is that we have the space definied for the ObjectNamePrefix for the CRS extension.
For the Affix in the AppSourceCop.json we have the affixes defined without the space.

For us it would be useful if the spaces between the affix and the name were ignored.

**Comment by rvanbekkum** [comment]
If needed, the same for `_` (underscores) 😊
We tend to have an underscore between our affix and the actual name.

**Comment by Arthurvdv** [comment]
Thanks for the feedback!

I now see that I had not foreseen that non-alphanumeric characters could be included in the prefix. This should now bee resolved in the updated ([v0.30.27](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.27) pre-release version of the LinterCop.

**Comment by Arthurvdv** [comment]
Thank you for your patience on this rule and reporting this.

I see the rule had a logic error and assumed the interface could to start with an capital 'I' when no Prefix/Affixes where set in the AppSourceCop.json file. I think I had a bit to much focus on the prefixing of the object names 😅

This would be a great rule to have the Test Automation of @christophstuber, but for the love of god I can't seem to get this working in my VS Code. I'll have to look into this in the next weeks.
